zstd/decoder.go

```diff
@@ -918,18 +919,18 @@ decodeStream:
 				println("next block returned error:", err)
 			}
 			dec.err = err
-			dec.checkCRC = nil
+			dec.hasCRC = false
 			if dec.Last && frame.HasCheckSum && err == nil {
 				crc, err := frame.rawInput.readSmall(4)
-				if err != nil {
+				if len(crc) < 4 {
```

Comment:
In case it's not obvious, this way of checking for the error triggers BCE in the else branch.

Reply:
No problem, but please make sure that `err` isn't somehow nil. If it is, set it to `io.ErrUnexpectedEOF`.

Reply:
Done!